toolchain/parse/node_kind.def

```diff
@@ -681,6 +681,35 @@ CARBON_PARSE_NODE_KIND_BRACKET(InterfaceDecl, InterfaceIntroducer,
                                CARBON_TOKEN(Semi)
                                    CARBON_IF_ERROR(CARBON_TOKEN(Interface)))
 
+// `impl ... as`:
+//     ImplIntroducer
+//     _repeated_ _external_: AccessModifierKeyword or DeclModifierKeyword
+//         _external_: ImplicitParamList
+//       ImplForall
+//     _optional_
+//     _optional_ _external_: expression
+//     ImplAs
+//     _external_: expression
+//   ImplDefinitionStart
+//   _external_: declarations
+// ImplDefinition
+//
+// The above is the structure for a definition; for a declaration,
+// ImplDefinitionStart and later nodes are removed and replaced by
+// ImplDecl.
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(ImplIntroducer, 0, CARBON_TOKEN(Impl))
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(ImplForall, 1, CARBON_TOKEN(Forall))
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(ImplAs, 0,
+                                   CARBON_TOKEN(As)
+                                       CARBON_IF_ERROR(CARBON_ANY_TOKEN))
```

Comment:
Is this true of the current `ImplAs`? It looks like you're only emitting it after finding the `As` token.